src/Harmony.php

```diff
@@ -72,20 +74,30 @@ public function getMiddleware(string $id): ?MiddlewareInterface
 
     public function addMiddleware(MiddlewareInterface $middleware, ?string $id = null): Harmony
     {
-        $this->middleware[] = [
+        return $this->add([
             "id" => $id,
             "middleware" => $middleware,
-        ];
-
-        return $this;
+        ]);
     }
 
     public function addCondition(ConditionInterface $condition, callable $callableOnSuccess): Harmony
     {
-        $this->middleware[] = [
+        return $this->add([
             "condition" => $condition,
             "middleware" => $callableOnSuccess,
-        ];
+        ]);
+    }
+
+    /**
+     * @param array<string, mixed> $middlewareArray
+     */
+    protected function add(array $middlewareArray): Harmony
+    {
+        if ($this->injectNewMiddlewarePosition === null) {
```

Comment:
Won't this solution lose track of the current middleware position in case of nested conditions?

Reply:
You've got a point there. Nested conditions does not work with my fix. On the other hand, the current version does not support anything after a condition.